packages/kit/src/runtime/client/client.js

```diff
@@ -1234,8 +1234,14 @@ export function create_client({ target, session, base, trailing_slash }) {
 			});
 		},
 
-		_hydrate: async ({ status, error, nodes, params, routeId }) => {
+		_hydrate: async ({ status, error, nodes, routeId }) => {
 			const url = new URL(location.href);
+			const params = get_navigation_intent(url)?.params || {};
+			Object.keys(params).forEach((key) => {
+				// Can't just use decodeURIComponent here because it's already _mostly_ decoded and that
+				// would double-decode. So we do roughly what "devalue" does.
+				params[key] = params[key].replace(/%2f/gi, '/').replace(/%3f/gi, '?');
```

Comment:
if we were going to do this, it should use the same function we use on the server: https://github.com/sveltejs/kit/blob/52fc5b367f34c79cba90ae7114db02b5cfbf9a59/packages/kit/src/runtime/server/utils.js#L14

Reply:
That makes sense, I'll update the PR this evening.

Reply:
See https://github.com/sveltejs/kit/commit/d02f1f25ac8acb29e21a06b94418c333928fb9bb

Reply:
Updated. Looks like `get_navigation_intent` now takes care of decoding the params, so I was able to simplify that part.